source/OutfitterPanel.cpp

```diff
@@ -315,186 +355,353 @@ double OutfitterPanel::DrawDetails(const Point &center)
+	int mountsFree = playerShip->Attributes().Get("turret mounts");
+	if(mountsNeeded && !mountsFree)
+		errors.emplace_back("This weapon is designed to be installed on a turret mount, "
+			"but your ship does not have any unused turret mounts available.");
+
+	int gunsNeeded = -selectedOutfit->Get("gun ports");
+	int gunsFree = playerShip->Attributes().Get("gun ports");
+	if(gunsNeeded && !gunsFree)
+		errors.emplace_back("This weapon is designed to be installed in a gun port, "
+			"but your ship does not have any unused gun ports available.");
+
+	if(selectedOutfit->Get("installable") < 0.)
+		errors.emplace_back("This item is not an outfit that can be installed in a ship.");
+
+	// For unhandled outfit requirements, show a catch-all error message.
+	if(errors.empty())
+		errors.emplace_back("You cannot install this outfit in your ship, "
+			"because it would reduce one of your ship's attributes to a negative amount.");
+
+	// Return the errors in the appropriate format.
 	if(errors.empty())
 		return true;
-	else if(errors.size() == 1)
+
+	if(errors.size() == 1)
 		return errors[0];
-	else
+
+	string errorMessage = "There are several reasons why you cannot buy this outfit:\n";
+	for(const auto & error : errors)
+		errorMessage += "- " + error + "\n";
+
+	return errorMessage;
+}
+
+
+
+// Used in both the Sell and the Uninstall contexts. Are we able to remove Outfits from
+// the selected ships?
+// If not, return the reasons why not.
+ShopPanel::TransactionResult OutfitterPanel::CanSellOrUninstall(const string &verb) const
+{
+	if(!planet || !selectedOutfit)
+		return "No outfit selected.";
+
+	if(!playerShip)
+		return "No ship selected.";
+
+	if(selectedOutfit->Get("map"))
+		return "You cannot " + verb + " maps. Once you buy one, it is yours permanently.";
+
+	if(HasLicense(selectedOutfit->TrueName()))
+		return "You cannot " + verb + " licenses. Once you obtain one, it is yours permanently.";
+
+	for(const Ship *ship : playerShips)
+		if(ShipCanRemove(ship, selectedOutfit))
+			return true;
+
+	// Something is being blocked from being sold or we don't have any.
+	// Check if we have one available or not, and then whether other outfit depend on it,
+	// if so dependent outfits must be uninstalled first.
+	// Collect the reasons that the outfit can't be sold or uninstalled.
+	vector<string> errors;
+	bool hasOutfit = false;
+	for(const Ship *ship : playerShips)
+		if(ship->OutfitCount(selectedOutfit))
+		{
+			hasOutfit = true;
+			break;
+		}
+
+	if(!hasOutfit)
+		return "You don't have any of these outfits to " + verb + ".";
+
+	for(const Ship *ship : playerShips)
+		if(ship->OutfitCount(selectedOutfit))
+			for(const pair<const char *, double> &it : selectedOutfit->Attributes())
+				if(ship->Attributes().Get(it.first) < it.second)
+				{
+					bool detailsFound = false;
+					for(const auto &sit : ship->Outfits())
+						if(sit.first->Get(it.first) < 0.)
+						{
+							errors.push_back("You cannot " + verb + " this outfit, "
```

Comment:
I don't know that I have an example of many outfits, but here's many conditions and many ships:

![image](https://github.com/user-attachments/assets/41bfd749-adbd-4a28-b7fd-de9bf0632f81)

Reply:
Do you want me to try to compile more concise, yet still exhaustive, output?

E.g. in the above screenshot, there are two reasons driving the same response for both of the ships...
we could:
1. you cannot sell <outfit> from <ship> because (reasons).
2. you cannot sell <outfit> from <ships> because (reason).
...

Thoughts?

Reply:
Here's what it looks like with my latest commit based on your feedback:

| One ship selected | Multiple ships same problems | Multiple ships diff probs | Multiple ships only one has item to sell |
|--------|--------|--------|--------|
| ![image](https://github.com/user-attachments/assets/8d9061de-b193-49c2-92d6-7e67ef6394cd)| ![image](https://github.com/user-attachments/assets/69ba6f60-99aa-4fc1-8dc2-4f34997d7a99)| ![image](https://github.com/user-attachments/assets/715c8114-7bbb-40ad-8d4a-bae97dd2a3c4)| ![image](https://github.com/user-attachments/assets/1cb0d449-8543-4ea0-9c39-cad40add1ab7)|